geo/benches/sweep_line_intersection.rs

```diff
@@ -54,116 +63,78 @@ fn brute_force_intersections(lines: &[Line<f64>]) -> Vec<(Line<f64>, Line<f64>)>
     result
 }
 
-// Benchmark performance comparison across different dataset sizes
-fn bench_performance_comparison(c: &mut Criterion) {
-    let mut group = c.benchmark_group("Performance Comparison");
-    group.sample_size(10);
-
+// Benchmark with "dense" case - lines with many intersections.
+// When intersections are dense, the sweep algorithm has less of an advantage vs. brute force.
+fn bench_dense_line_intersections(c: &mut Criterion) {
     let mut rng = StdRng::seed_from_u64(42);
-
-    // Test key sizes: crossover point, medium, and large datasets
-    for &n in &[1000, 5000, 10000] {
-        let lines = generate_random_lines(n, &mut rng);
+    for (n, sample_size, expected_intersections) in [
+        (10, None, 7),
+        (100, None, 861),
+        (1_000, Some(50), 91_898),
+        (10_000, Some(10), 8_570_900),
+    ] {
+        let mut group = c.benchmark_group(format!("Random dense lines ({n} lines)"));
+        if let Some(sample_size) = sample_size {
+            group.sample_size(sample_size);
+        }
+        let lines = generate_random_lines(n, 4.0, &mut rng);
 
         // Brute force approach
-        group.bench_function(format!("brute_force_n{n}"), |b| {
+        group.bench_function("brute_force", |b| {
             b.iter(|| {
-                black_box(brute_force_intersections(&lines));
+                let intersections = black_box(brute_force_intersections(&lines));
+                assert_eq!(intersections.len(), expected_intersections);
             });
         });
 
         // Sweep line algorithm
-        group.bench_function(format!("sweep_n{n}"), |b| {
+        group.bench_function("sweep", |b| {
             b.iter(|| {
                 let intersections: Vec<_> =
                     NewSweepIntersections::<_>::from_iter(lines.iter().cloned()).collect();
-                black_box(intersections);
+                assert_eq!(intersections.len(), expected_intersections);
             });
         });
-    }
-
-    group.finish();
-}
 
-// Benchmark with "dense" case - many lines in small area
-// The new algorithm should be faster here
-fn bench_dense_line_intersections(c: &mut Criterion) {
-    let mut group = c.benchmark_group("Dense Line Intersections");
-    group.sample_size(10);
-
-    // Create a grid of lines that will have many intersections
-    // expected intersections = (n / 2) * (n / 2);
-    let n = 1000;
```

Comment:
Note - I replaced the previous "dense" and "sparse" logic with something tunable that can be used for both sparse and dense use cases.

However, if you think there's value in this previous approach - e.g. like maybe it's more realistic than the random lines I'm now generating, I could be convinced to revert that part.